src/ansys/mapdl/core/mapdl_grpc.py

```diff
@@ -881,8 +881,8 @@ def _launch(self, start_parm, timeout=10):
                 self.prep7()
                 success = True
                 break
-            except:
-                pass
+            except MapdlRuntimeError:
+                warn("PyMAPDL is taking longer than expected to connect to the server.")
```

Comment:
Couple of things:

* We are not throwing a **runtime** error but a warning while waiting for timeout... I think it's valid
* If it fails, later down the road, the exception will be raised,
* Better say something is happening than just "the void" IMO - having some warning raised or info message is a good solution.

Reply:
For completeness: https://stackoverflow.com/questions/22661745/why-only-one-warning-in-a-loop

Bottom line, the warning message should only be shown once even though it might get triggered N times

Reply:
Ok... I apologise. I did not read it properly at the beginning.

It is not bad idea the warning... and because this is not following the standard connection procedure, the `MapdlRuntimeError` is fine. Just please add a `time.sleep(1)` before the warning, because otherwise we are going to print warning blazingly fast.

```py
# may need to wait for viable connection in open_gui case
tmax = time.time() + timeout
success = False
icounter = 0
while time.time() < tmax:
    try:
        self.prep7()
        success = True
        break

    except MapdlRuntimeError:
        icounter += 1
        time.sleep(1)

        if (icounter % 5) == 0:
            warn("PyMAPDL is taking longer than expected to connect to the server.")
```

Reply:
That I agree if they are going to be executed many times. Thanks for the suggestion @germa89

Reply:
> https://stackoverflow.com/questions/22661745/why-only-one-warning-in-a-loop

OK I didnt' know that... then... just add the `sleep` call and open the issue.

Reply:
Good idea, I can add it

Reply:
Yep - the sleep alone should work :)

Reply:
Applied in [0ddb90b](https://github.com/ansys/pymapdl/pull/3474/commits/0ddb90b7cf3fb6bb38bfcedc826f8ac0b55a0322)